### ISSUE TITLE
Additional search option

### ISSUE THREAD

**Issue by GiantCrocodile** [GetPublii/Publii#321]
I would suggest another search option (although I'm unsure which one). Reasons:

1. DocSearch is nice but not fitting to a blog/static site - it's focus is source code docs, according to what I understood.
2. Googles custom search is preferred in my point of view because of privacy and monopoly reasons.

I would like to have a search which is either javascript based (locally in your browser, I think there are frameworks for it but I think it may be too heavy for the site) or some kind of pre-generated search results: searching by title, tags, author, [...].

**Comment by dzmitry-lahoda** [comment]
Add one of jamstack-static searches(not all searches in image are such, but kind of):
![image](https://user-images.githubusercontent.com/757125/53176299-d7343900-35fe-11e9-8404-46dcdf5b79be.png)
I will donate to 10 USD as soon as it will be added.

**Comment by dzmitry-lahoda** [comment]
`gatsby` has jamstack searched plugins https://www.gatsbyjs.org/plugins/?=search

**Comment by aadrian** [comment]
Please add **configurable search endpoint** settings so that the community can quickly write examples how to integrate with various other search solutions, or even to setup one such search endpoint (e.g. based on [Apache Solr](http://lucene.apache.org/solr/ or https://www.elastic.co/products/elasticsearch) 

Thank you.

**Comment by wuniversales** [comment]
I would like to be able to add searches in this way.

`https://duckduckgo.com/?q=site:my.website + search`

I think this would be much better for everyone :)

**Comment by wuniversales** [comment]
@dziudek Well, I have tested the changes I have made, and it just works. #838
I have focused on leaving the final result of the web as clean as possible (Avoid unnecessary code, and avoid the use of javascript by the client).

However I have needed some javascript to show the external searches with results from the user's website.

All these changes I am making because I need them for my websites, since I program them for myself, I contribute them to the publishing community in order to be implemented in future versions.

Greetings, I hope they are implemented as soon as possible. :)
